server/src/main/java/org/elasticsearch/index/similarity/SimilarityProviders.java

```diff
@@ -56,74 +56,46 @@
-        // Is there a better replacement for d and p which use the binomial model?
-        legacyModels.put("be", "g");
-        legacyModels.put("d", "ine");
-        legacyModels.put("p", "ine");
-        LEGACY_BASIC_MODELS = unmodifiableMap(legacyModels);
-
-        Map<String, AfterEffect> effects = new HashMap<>();
-        effects.put("b", new AfterEffectB());
-        effects.put("l", new AfterEffectL());
-        AFTER_EFFECTS = unmodifiableMap(effects);
-
-        Map<String, String> legacyEffects = new HashMap<>();
-        // l is simpler than b, so this should be a better replacement for "no"
-        legacyEffects.put("no", "l");
-        LEGACY_AFTER_EFFECTS = unmodifiableMap(legacyEffects);
-    }
-
-    private static final Map<String, Independence> INDEPENDENCE_MEASURES;
-    static {
-        Map<String, Independence> measures = new HashMap<>();
-        measures.put("standardized", new IndependenceStandardized());
-        measures.put("saturated", new IndependenceSaturated());
-        measures.put("chisquared", new IndependenceChiSquared());
-        INDEPENDENCE_MEASURES = unmodifiableMap(measures);
-    }
+    private static final Map<String, BasicModel> BASIC_MODELS = Map.of(
+            "g", new BasicModelG(),
+            "if", new BasicModelIF(),
+            "in", new BasicModelIn(),
+            "ine", new BasicModelIne());
+    // TODO: be and g and both based on the bose-einstein model.
+    // Is there a better replacement for d and p which use the binomial model?
+    private static final Map<String, String> LEGACY_BASIC_MODELS = Map.of(
+            "be", "g",
+            "d", "ine",
+            "p", "ine");
+    private static final Map<String, AfterEffect> AFTER_EFFECTS = Map.of(
+            "b", new AfterEffectB(),
+            "l", new AfterEffectL());
+    // l is simpler than b, so this should be a better replacement for "no"
+    private static final Map<String, String> LEGACY_AFTER_EFFECTS = Map.of("no", "l");
+
+    private static final Map<String, Independence> INDEPENDENCE_MEASURES =  Map.of(
+            "standardized", new IndependenceStandardized(),
+            "saturated", new IndependenceSaturated(),
+            "chisquared", new IndependenceChiSquared());
 
     private static final Map<String, Distribution> DISTRIBUTIONS;
     private static final Map<String, Lambda> LAMBDAS;
 
     static {
-        Map<String, Distribution> distributions = new HashMap<>();
-        distributions.put("ll", new DistributionLL());
-        distributions.put("spl", new DistributionSPL());
-        DISTRIBUTIONS = unmodifiableMap(distributions);
-
-        Map<String, Lambda> lamdas = new HashMap<>();
-        lamdas.put("df", new LambdaDF());
-        lamdas.put("ttf", new LambdaTTF());
-        LAMBDAS = unmodifiableMap(lamdas);
+        DISTRIBUTIONS = Map.of("ll", new DistributionLL(), "spl", new DistributionSPL());
+
+        LAMBDAS = Map.of("df", new LambdaDF(), "ttf", new LambdaTTF());
```

Comment:
move initialization to the declaration and remove the static block, like you did for other constants?